### ISSUE TITLE
Update README

### ISSUE THREAD

**Issue by swill** [ctessum/go.clipper#6]
Update the Readme to include build and usage examples.

**Comment by ctessum** [comment]
Contributions welcome!

**Comment by swill** [comment]
I would love to, but I can't figure out how to make it work (which is the problem).

For example, how do you create a `clipper.Path` outside the package?  It is made up of `clipper.Path` => `[]*clipper.IntPoint` => `cInt`.  But `clipper.cInt` is not exported by the package (lowercase first letter), so you get the following when you try to create: `cannot refer to unexported name clipper.cInt`.

Also, how do you use `int64` vs `int32`?

So once I get things working and understand how the package is expected to work, I can start helping and contributing.  But the lack of a readme really makes that hard...

**Comment by ctessum** [comment]
I actually haven't extensively used the library myself, because after porting it I realized that it only works with integers and I need to be working with floats. So I've been focusing on fixing the floating point math-related errors in [polyclip](https://github.com/ctessum/polyclip-go) instead.

**Comment by swill** [comment]
Fair enough.  👍  I have started playing with the library, so I will open PRs for you with my improvements.

I also use floating points in my application (builder.swillkb.com), but I have been having a lot of problems with polyclip, so I am actively looking to rewrite my application to remove the dependency on polyclip because it has too many problems.

I am planning to *fake* floating point support in go.clipper with the following type of functionality.

```
const (
	PRECISION float64 = 1000
)

type Point struct {
	X float64
	Y float64
}

type Points []Point

func (ps Points) ToPath() clipper.Path {
	p := make(clipper.Path, 0)
	for i := range ps {
		p = append(p, &clipper.IntPoint{clipper.CInt(ps[i].X * PRECISION), clipper.CInt(ps[i].Y * PRECISION)})
	}
	return p
}

func FromPath(cp clipper.Path) Points {
	p := make(Points, 0)
	for i := range cp {
		p = append(p, Point{float64(cp[i].X) / PRECISION, float64(cp[i].Y) / PRECISION})
	}
	return p
}
```
(Note: I will create a pull request for the exporting of `CInt` by the `clipper` package so external projects can use the `clipper` library)

This approach is not perfect, but it lets me get to `+/- 0.001mm` precision in my application, which is good enough for my use case.

I have only done initial testing right now, but I plan to test a rewrite to remove polyclip from my application and depend on go.clipper instead in a feature branch, so we will see how it goes.

**Comment by swill** [comment]
Here is the PR I mentioned in my last post: #7

**Comment by ctessum** [comment]
Sure. An additional option could be to try my branch of [polyclip-go](https://github.com/ctessum/polyclip-go) to see if the fixes I have made there also fix your problems. If your problems still exist, if you open an issue there with an example I would be happy to work on it.

**Comment by ctessum** [comment]
I should mention that I did already try a similar approach to what you mention above to get floating points to work with clipper, but was not successful. The problem I ran into is that with geographical data, the data can be in different units that give values ranging across ~8 orders of magnitude, so it's not really possible to select a constant PRECISION that works in all cases. You may not have that problem when working with CAD data though.

**Comment by swill** [comment]
I wrote a small monitoring tool called [`slackd`](https://github.com/swill/slackd) specifically to get a better idea why polyclip was failing, so I have a HUGE number of failed unions logged.  I parsed out 101 cases where polyclip failed to create a union and have been using them for testing.

```
akavel/polyclip-go => 101/101 FAIL
ctessum/polyclip-go => 101/101 FAIL
swill/go.clipper => 6/101 FAIL (with my float -> int -> float transition)
```

So this is promising.  For my use case, I am working with numbers between `0 - 500mm` (for the most part) and a precision of `+/-0.001mm` is good enough for my use case, so that means the largest number I have to work with (in theory) is 500,000.  So I think this will work for me...

Right now I just have a json dump of the different union cases, so it may not be very useful for you, but I have attached it anyway...

[test_cases.json.zip](https://github.com/ctessum/go.clipper/files/386977/test_cases.json.zip)

**Comment by swill** [comment]
Here is my very basic testing so far.  All the errors are when doing a union of a switch and a stabilizer, so the tests are all almost identical.  About 80% of the time this union works, but periodically it fails and here are some of the failures.  I have found that the exact same points will succeed and then if moved slightly on the X axis, they will then fail.  I have not found the exact reason for the failures, but I have not dug into it too deeply yet.

https://github.com/swill/poly_op_tests

**Comment by ctessum** [comment]
Thanks. I'll take a look at this.

**Comment by swill** [comment]
Last night I ripped out polyclip from my application and replaced it with go.clipper using the `precision` concept I described earlier.  So far I am VERY happy with the results.  I am able to do things that I was having a really hard time doing with polyclip by utilizing a combination of `clipper.PftEvenOdd` and `clipper.PftNonZero`.  I am also getting much better results when doing`clipper.CtDifference` operations which produce multiple polygons with holes in them.

All said, I am looking forward to working with this library more and I am very happy with the results of my initial implementation.  Thanks for taking the time to port this to Go.  👍  I will continue to send you PRs if I fix other stuff.
